database/sqlite/sqlite_aclk_node.c

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include "sqlite_functions.h"
+#include "sqlite_aclk_node.h"
+
+#ifdef ENABLE_ACLK
+#include "../../aclk/aclk_charts_api.h"
+#endif
+
+void sql_build_node_info(struct aclk_database_worker_config *wc, struct aclk_database_cmd cmd)
+{
+    UNUSED(cmd);
+
+#ifdef ENABLE_ACLK
+    struct update_node_info node_info;
+
+    rrd_wrlock();
+    node_info.node_id = strdupz(wc->node_id);
+    node_info.claim_id = is_agent_claimed();
+    node_info.machine_guid = strdupz(wc->host_guid);
+    node_info.child = (wc->host != localhost);
+    now_realtime_timeval(&node_info.updated_at);
+
+    RRDHOST *host = wc->host;
+
+    node_info.data.name = strdupz(host->hostname);
+    node_info.data.os = strdupz(host->os);
+    node_info.data.os_name = strdupz(host->system_info->host_os_name);
+    node_info.data.os_version = strdupz(host->system_info->host_os_version);
+    node_info.data.kernel_name = strdupz(host->system_info->kernel_name);
+    node_info.data.kernel_version = strdupz(host->system_info->kernel_version);
+    node_info.data.architecture = strdupz(host->system_info->architecture);
+    node_info.data.cpus = str2uint32_t(host->system_info->host_cores);
+    node_info.data.cpu_frequency = strdupz(host->system_info->host_cpu_freq);
+    node_info.data.memory = strdupz(host->system_info->host_ram_total);
+    node_info.data.disk_space = strdupz(host->system_info->host_disk_space);
+    node_info.data.version = strdupz(VERSION);
```

Comment:
especially here strdupz is wasteful. just point to it directly the `VERSION` isn't going anywhere nor is it being modified by other thread

Reply:
> an additional small nit picky comment would be should those files and functions be called sqlite? I know they are called out of your event loop but same time this particular part seems it has nothing to do with database

True -- could be just "node_info" or something